src/lib.rs

```diff
@@ -1,58 +1,19 @@
 #![doc(html_root_url = "https://docs.rs/rfc2047-decoder/0.1.2")]
 
-use std::{error, fmt, result};
-
 mod evaluator;
 mod lexer;
 mod parser;
 
-/// Errors can come from the lexer or the parser.
-/// Lexer errors are related to invalid syntaxes.
-/// Parser and Evaluator errors are related to decoding issues.
-#[derive(Debug)]
+#[derive(thiserror::Error, Debug)]
 pub enum Error {
-    RunLexerError(lexer::Error),
-    RunParserError(parser::Error),
-    RunEvaluatorError(evaluator::Error),
+    #[error(transparent)]
+    Lexer(#[from] lexer::Error),
+    #[error(transparent)]
+    Parser(#[from] parser::Error),
+    #[error(transparent)]
+    Decode(#[from] evaluator::Error),
```

Comment:
Why no keeping `Evaluator`? In order to stay consistent with error and file name.

Reply:
> Why no keeping Evaluator?

What do you mean with that?

Reply:
Ooooh, you mean the enum-entry! Hm.... I dunno I think renaming `evaluator` to `decoder` might be more suitable, what do you think?

Reply:
Lexer / parser / evaluator seems to be common naming for such operations. I took inspiration from various articles, in particular this one https://accu.org/journals/overload/26/147/balaam_2565/.

Reply:
done